server/src/main/java/keywhiz/service/permissions/AllowClientSecretCreationPermissionCheck.java

```diff
@@ -0,0 +1,18 @@
+package keywhiz.service.permissions;
+
+import keywhiz.api.model.Client;
+import keywhiz.api.model.Secret;
+
+public class AllowClientSecretCreationPermissionCheck implements PermissionCheck {
+  @Override
+  public boolean isAllowed(Object source, String action, Object target) {
+    return false;
+  }
+
+  @Override
+  public boolean isAllowedForTargetType(Object source, String action, Class<?> targetType) {
```

Comment:
I'm probably missing a lot of context, but are we not use `action` at all?

Reply:
Oh man! I can't believe I missed that. Yes, it should totally include action. Derp derp derp